pkg/worker/lifecycle.go

```diff
@@ -507,3 +517,19 @@ func (s *Worker) createOverlay(request *types.ContainerRequest, bundlePath strin
 func (s *Worker) isBuildRequest(request *types.ContainerRequest) bool {
 	return request.SourceImage != nil
 }
+
+func (s *Worker) watchForOOMKills(ctx context.Context, containerId string) {
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			if c, ok := s.containerInstances.Get(containerId); ok && c.isOOMKilled() {
+				log.Printf("<%s> - [WARNING] A process in the container was killed due to out-of-memory conditions.\n", containerId)
```

Comment:
I think this needs to use the logger on the container instance, otherwise it won't show up for users

Reply:
Replaced the print with sending to the output channel.